### ISSUE TITLE
Git + R nirvana: how to get there

### ISSUE THREAD

**Issue by jennybc** [ropensci/unconf18#22]
The tools most of us use to accomplish Git/GitHub magic **from R** are the RStudio IDE and the [git2r](https://github.com/ropensci/git2r) package (part of the rOpenSci org).

Under the hood, these exploit different tools to enact Git operations: system Git (RStudio IDE) and libgit2 (git2r). This means that various aspects of Git configuration can be good to go for one but not the other. This is mostly about configuring credentials for Git remotes, e.g. setting up SSH keys.

I've done a fair amount of testing and documenting for [Happy Git](http://happygitwithr.com). But I think initial setup could become even easier and better documented with respect to tricky bits, such as using passphrase-protected SSH keys on Windows. I'd love to stress-test and improve setup instructions for Git so that more people have more success across Mac/Windows/Linux for command line Git (--> RStudio IDE) and git2r (--> devtools, usethis, etc.).

**Comment by maurolepore** [comment]
> But I think initial setup could become even easier and better documented with respect to tricky bits...

I totally agree. Considering the key role of Git and GitHub in data science workflows, and how nicely integrated these tools are with RStudio, I think the resources for people to learn remain scarce. For example, there is only one RStudio webinar on GitHub and RStudio ([Managing Change Part 2](https://www.rstudio.com/resources/webinars/)), which -- I believe -- doesn't cover SSH keys, and was recorded before RStudio included a terminal.

**Comment by maurolepore** [comment]
A little off-topic but still related to git2r. A tool to [Automatically generate list of contributors](https://github.com/r-lib/devtools/issues/1279) has already been packaged up?

>  I have local code to achieve this and will package up at some point
-- @hadley
